[RFC] ipq806x: use qcom-ipq8064.dtsi from upstream

adschm opened the pull request:
Though a qcom-ipq8064.dtsi file exists upstream, we still do overwrite
it with a full version of our own in the ipq806x target. About half of
the contents of our file are upstream content, the other half are local
improvements.

To prevent us from having a lot of code maintained twice in parallel,
this adjusts the target to use the upstream qcom-ipq8064.dtsi with
trivial adjustments in patches, and move all specific definitions to
the qcom-ipq8064-v1.0.dtsi which we still maintain locally.

This should get us the best of both worlds.

adschm committed:
ipq806x: use qcom-ipq8064.dtsi from upstream

Though a qcom-ipq8064.dtsi file exists upstream, we still do overwrite
it with a full version of our own in the ipq806x target. About half of
the contents of our file are upstream content, the other half are local
improvements.

To prevent us from having a lot of code maintained twice in parallel,
this adjusts the target to use the upstream qcom-ipq8064.dtsi. Our
local changes are arranged into three patches, the first pulling a
commit from upstream, the second doing a few small adjustments, and
the third adding all additional stuff.

This should get us the best of both worlds.

The property "ports-implemented" on sata@29000000 is moved to
2nd-level DTSI files as kernel defines it there as well.

While at, rename 080-ARM-dts-qcom-add-gpio-ranges-property.patch to
include the kernel version where it's added upstream.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
@Ansuel @mans0n 
This idea struck me when I thought about #3160

I'm not sure whether it's a good one yet, so I'd be happy about honest comments here. I'd be ready to dump everything if you say it's not really an improvement.

adrianschmutzler commented on target/linux/ipq806x/patches-5.4/100-ipq8064-dtsi-tweaks.patch:
```diff
@@ -0,0 +1,206 @@
+ #include <dt-bindings/soc/qcom,gsbi.h>
+ #include <dt-bindings/interrupt-controller/arm-gic.h>
++#include <dt-bindings/mfd/qcom-rpm.h>
++#include <dt-bindings/clock/qcom,rpmcc.h>
```
I'm not sure where these two lines are actually needed, I suspect they can be moved to 1.0.dtsi at least ...

mans0n commented:
From my view, this just splits local ipq8064.dtsi into three: upstream ipq8064.dtsi, local ipq8064-v1.0.dtsi, and local ipq8064-dtsi-tweaks.patch. And it still overwrites upstream ipq8064-v1.0.dtsi. Personally I do not see much gain from this...

adrianschmutzler commented:
I've changed this conceptionally so all changes to qcom-ipq8064.dtsi are maintained in patches, which seems cleaner to me than the previous approach. Leveraged by that, a second patch now disentangles the other DTSI files, so one can use them to target individual subsets of devices more easily.

Ansuel commented on target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8065.dtsi:
```diff
@@ -49,68 +55,53 @@
 	};
 
 	opp-1400000000 {
-		opp-hz = /bits/ 64 <1400000000>;
```
I can't really find where this is reproduced... This is needed to make the scaling works 
Same for opp-supported and clock

adrianschmutzler replied:
These definitions are already in the ipq8064.dtsi

Since the node name is the same, we are referencing these very blocks and are just _updating_ them.
That's also why we need to use delete-node above.

https://github.com/openwrt/openwrt/blob/master/target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8064.dtsi#L145

Ansuel replied:
you are right. sorry

mans0n commented on target/linux/ipq806x/files-5.4/arch/arm/boot/dts/qcom-ipq8064-db149.dts:
```diff
@@ -14,8 +14,7 @@
-	alias {
-		serial0 = &gsbi2_serial;
+	aliases {
 		mdio-gpio0 = &mdio0;
```
If we are going to drop serial0, then IMO we can also drop mdio-gpio0, which is for [mdio-gpio](https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/mdio-gpio.txt#L17).

adrianschmutzler commented:
I merged the trivial/less controversial stuff to master, and updated the rest here based on the recent state.

adschm committed:
ipq806x: use qcom-ipq8064-v1.0.dtsi from upstream

This uses upstream qcom-ipq8064-v1.0.dtsi and modifies it by patches
instead of keeping a local version. As a consequence:

- we use a part of the shared definitions there and update device
  DTS files accordingly
- we move additional stuff from our local v1.0.dtsi to the patch
- we drop partitions, LEDs and keys from the file as we will
  implement them differently anyway

Like with the previous patch, this follows the idea that a diff
from upstream might be easier to handle than a big file of our
own with different distribution pattern of properties.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
I've merged the remaining simple adjustments, rebased, and added a patch for dealing with upstream qcom-ipq8064-v1.0.dtsi as well.

adrianschmutzler commented:
I still consider it an improvement, so I merged it.
